pkg/api/server/v1alpha2/auth/rbac.go

```diff
@@ -38,22 +39,41 @@ import (
 // This checks RBAC permissions in the `results.tekton.dev` group, and assumes
 // checks are done at the namespace
 type RBAC struct {
-	authn authnclient.AuthenticationV1Interface
-	authz authzclient.AuthorizationV1Interface
+	impersonation bool
+	authn         authnclient.AuthenticationV1Interface
+	authz         authzclient.AuthorizationV1Interface
 }
 
-func NewRBAC(client kubernetes.Interface) *RBAC {
-	return &RBAC{
+type Option func(*RBAC)
+
+func NewRBAC(client kubernetes.Interface, options ...Option) *RBAC {
+	rbac := &RBAC{
 		authn: client.AuthenticationV1(),
 		authz: client.AuthorizationV1(),
 	}
+	for _, option := range options {
+		option(rbac)
+	}
+	return rbac
 }
 
 func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) error {
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
 		return status.Error(codes.Unauthenticated, "unable to get context metadata")
 	}
+
+	// Parse Impersonation header if the feature is enabled
+	var i *impersonation.Impersonation
```

Comment:
Please use a more descriptive variable name here.

```suggestion
	var impersonator *impersonation.Impersonation
```

Reply:
updated